Chaincase.Common/Global.cs

```diff
@@ -343,7 +343,68 @@ public async Task<bool> WaitForInitializationCompletedAsync(CancellationToken ca
         private protected CancellationTokenSource ResumeCts { get; set; } = new CancellationTokenSource();
         private protected bool IsResuming { get; set; } = false;
 
-        public async Task OnResuming()
+        public void HandleRemoteNotification()
+        {
+            if (_walletManager?.SleepingCoins is { } && _torManager?.State != TorState.Started && _torManager.State != TorState.Connected)
+            {
+                _ = ResumeToCoinJoin();
+                // sleep instruction happens from iOS lifecycle
+            }
+        }
+
+        public async Task ResumeToCoinJoin()
+		{
+            if (IsResuming)
+            {
+                Logger.LogDebug($"{nameof(Global)}.ResumeToCoinJoin(): SleepCts.Cancel()");
+                SleepCts.Cancel();
+                return;
+            }
+
+            try
+            {
+                Resumed?.Invoke(this, null);
+                Logger.LogDebug($"{nameof(Global)}.ResumeToCoinJoin(): Waiting for a lock");
+                ResumeCts.Dispose();
+                ResumeCts = new CancellationTokenSource();
+                using (await LifeCycleMutex.LockAsync(ResumeCts.Token))
+                {
+                    IsResuming = true;
```

Comment:
if we Foreground from here, we want to start other services. This flag should not be set here.